lib/puppetserver/ca/config/puppet.rb

```diff
@@ -34,32 +34,46 @@ def self.parse(config_path)
 
         def initialize(supplied_config_path = nil)
           @using_default_location = !supplied_config_path
-          @config_path = supplied_config_path || user_specific_conf_file
+          @config_path = supplied_config_path || user_specific_puppet_config
 
           @settings = nil
           @errors = []
         end
 
+        def running_as_root?
+          @as_root ||= Puppetserver::Ca::Utils::Config.running_as_root?
+        end
```

Comment:
Why do we need this instead of calling `Puppetserver::Ca::Utils::Config.running_as_root?` directly? Is it just an expensive thing to check?

Reply:
I don't remember if it's expensive, though previously we were caching the end result of the call so I preserved that behavior.

Reply:
We were? It looks like we were previously calling `Puppetserver::Ca::Utils::Config.running_as_root?` just once in this file and not caching that value anywhere?

Reply:
Looks like we were caching the answer we got as a result of the conditional in which this was called. So we wouldn't have called this more than once.